solidity/test/DepositRedemptionTest.js

```diff
@@ -674,6 +674,80 @@ describe("DepositRedemption", async function() {
     })
   })
 
+  describe("transferAndRequestRedemption", async () => {
+    const sighash =
+      "0xb68a6378ddb770a82ae4779a915f0a447da7d753630f8dd3b00be8638677dd90"
+    const outpoint = "0x" + "33".repeat(36)
+    const valueBytes = "0x1111111111111111"
+    const keepPubkeyX = "0x" + "33".repeat(32)
+    const keepPubkeyY = "0x" + "44".repeat(32)
+    // Override redeemer output script for this test.
+    // No real reason here, it's just how we derived the below values.
+    const redeemerOutputScript = "0x160014" + "33".repeat(20)
+    let requiredBalance
+
+    before(async () => {
+      requiredBalance = depositValue
+    })
+
+    beforeEach(async () => {
+      await createSnapshot()
+      await testDeposit.setState(states.ACTIVE)
+      await testDeposit.setUTXOInfo(valueBytes, 0, outpoint)
+
+      // make sure there is sufficient balance to request redemption. Then approve deposit
+      await tbtcToken.resetBalance(requiredBalance, {from: owner})
+      await tbtcToken.resetAllowance(testDeposit.address, requiredBalance, {
+        from: owner,
+      })
+      await tbtcDepositToken.approve(testDeposit.address, tdtId, {
+        from: owner,
+      })
+    })
+
+    afterEach(async () => {
+      await restoreSnapshot()
+    })
+
+    it("updates state successfully and fires a RedemptionRequested event", async () => {
+      await testDeposit.setVendingMachineAddress(owner)
+      const blockNumber = await web3.eth.getBlockNumber()
+
+      await testDeposit.setSigningGroupPublicKey(keepPubkeyX, keepPubkeyY)
+
+      // the fee is 2.86331153 BTC
```

Comment:
Is this… Relevant?

Reply:
slightly, the test vector is directly copy from `requestRedemption`, since they use pretty much the exact same logic, I figured not changing the test scenario was appropriate (including comments)